internal/controllers/topology/cluster/blueprint_test.go

```diff
@@ -276,8 +289,60 @@ func TestGetBlueprint(t *testing.T) {
 					MachineHealthCheck:            machineHealthCheck,
 				},
 				MachineDeployments: map[string]*scope.MachineDeploymentBlueprint{},
+				MachinePools:       map[string]*scope.MachinePoolBlueprint{},
 			},
 		},
+		{
+			name: "Should read a ClusterClass with a MachinePoolClass",
+			clusterClass: builder.ClusterClass(metav1.NamespaceDefault, "class1").
+				WithInfrastructureClusterTemplate(infraClusterTemplate).
+				WithControlPlaneTemplate(controlPlaneTemplate).
+				WithWorkerMachinePoolClasses(mps...).
+				Build(),
+			objects: []client.Object{
+				infraClusterTemplate,
+				controlPlaneTemplate,
+				workerInfrastructureMachinePoolTemplate,
+				workerBootstrapTemplate,
+			},
+			want: &scope.ClusterBlueprint{
+				ClusterClass: builder.ClusterClass(metav1.NamespaceDefault, "class1").
+					WithInfrastructureClusterTemplate(infraClusterTemplate).
+					WithControlPlaneTemplate(controlPlaneTemplate).
+					WithWorkerMachinePoolClasses(mps...).
+					Build(),
+				InfrastructureClusterTemplate: infraClusterTemplate,
+				ControlPlane: &scope.ControlPlaneBlueprint{
+					Template: controlPlaneTemplate,
+				},
+				MachineDeployments: map[string]*scope.MachineDeploymentBlueprint{},
+				MachinePools: map[string]*scope.MachinePoolBlueprint{
+					"workerclass2": {
+						Metadata: clusterv1.ObjectMeta{
+							Labels:      map[string]string{"foo": "bar"},
+							Annotations: map[string]string{"a": "b"},
+						},
+						InfrastructureMachinePoolTemplate: workerInfrastructureMachinePoolTemplate,
+						BootstrapTemplate:                 workerBootstrapTemplate,
+					},
+				},
+			},
+		},
+		{
+			name: "Fails if ClusterClass has a MachinePoolClass referencing a BootstrapConfig that does not exist",
```

Comment:
Let's also add a case for 
`Fails if ClusterClass has a MachinePoolClass referencing a InfrastructureMachinePoolTemplate that does not exist`